desktop/core/src/desktop/api2.py

```diff
@@ -688,6 +692,155 @@ def share_document(request):
     'document': doc.to_dict()
   })
 
+import datetime
+import redis
+
+@api_error_handler
+@require_POST
+def handle_submit(request):
+  # Extract the task name and params from the request
+  try:
+    data = json.loads(request.body)
+    task_name = data.get('taskName')
+    task_params = data.get('taskParams')
+  except json.JSONDecodeError as e:
+    return JsonResponse({'error': str(e)}, status=500)
+
+  if task_name == 'document cleanup':
```

Comment:
I would recommend creating all the task names as constants in tasks.py and skip the spaces in values.
Something like
```
DOCUMENT_CLEANUP = 'DOCUMENT_CLEANUP'
```